crates/bevy_math/src/bezier.rs

```diff
@@ -0,0 +1,163 @@
+use std::ops::{Add, Mul};
+
+use glam::{Vec2, Vec3};
+
+/// Provides methods for sampling Cubic Bezier curves.
+pub trait CubicBezier {
+    /// Represents a coordinate in space, this can be 3d, 2d, or even 1d.
+    type Coord: Copy + Mul<f32, Output = Self::Coord> + Add<Self::Coord, Output = Self::Coord>;
+
+    /// Returns the four control points of the cubic Bezier curve.
+    fn control_points(&self) -> [Self::Coord; 4];
+
+    /// Evaluate the cubic Bezier curve at the parametric value `t`.
+    fn evaluate_at(&self, t: f32) -> Self::Coord {
+        let p = self.control_points();
+        p[0] * (1. - t).powi(3)
+            + p[1] * t * 3.0 * (1.0 - t).powi(2)
+            + p[2] * 3.0 * (1.0 - t) * t.powi(2)
+            + p[3] * t.powi(3)
+    }
+
+    /// Split the Bezier curve into `subdivisions`, and sample the position at each.
+    fn to_points(&self, subdivisions: i32) -> Vec<Self::Coord> {
+        (0..=subdivisions)
+            .map(|i| {
+                let t = i as f32 / subdivisions as f32;
+                self.evaluate_at(t)
+            })
+            .collect()
+    }
+}
+
+/// A 2-dimensional Bezier curve.
+#[derive(Default, Clone, Copy, Debug, PartialEq)]
+pub struct CubicBezier2d(pub [Vec2; 4]);
+
+impl CubicBezier for CubicBezier2d {
+    type Coord = Vec2;
+
+    fn control_points(&self) -> [Self::Coord; 4] {
+        self.0
+    }
+}
+
+impl CubicBezier2d {
+    /// Returns the point along the Bezier curve at the supplied parametric value `t`.
+    pub fn evaluate_at(&self, t: f32) -> Vec2 {
+        CubicBezier::evaluate_at(self, t)
+    }
+
+    /// Iterate over points in the bezier curve from `t = 0` to `t = 1`.
+    pub fn to_points(&self, subdivisions: i32) -> Vec<Vec2> {
+        CubicBezier::to_points(self, subdivisions)
+    }
+}
```

Comment:
why do you need to reimplement those from the trait?

Reply:
I added these delegates so users of the Bezier types don't need to import the trait, which I find to be pretty annoying as a user. The trait is really an implementation detail to reduce code duplication, and doesn't need to be used publicly.

Reply:
Not a fan, but why not. Then:
* could you mark them as `inline`?
* could you make sure the docs are the same between all imply and the trait?